manager/manager.go

```diff
@@ -693,6 +693,10 @@ func (m *manager) GetRequestedContainersInfo(containerName string, options v2.Re
 	for name, data := range containers {
 		info, err := m.containerDataToContainerInfo(data, &query)
 		if err != nil {
+			if err == memory.ErrDataNotFound {
+				klog.Warningf("Error getting data for container %s because of race condition", name)
```

Comment:
What makes you think that it's caused by race condition? Is there anything we could do to fix it?

Reply:
Sorry for the confusion, maybe I should have phrased it better. The race condition exists when the [metrics for all the containers are requested](https://github.com/google/cadvisor/blob/5d029c266a072305c0239990c714345b680f7d42/metrics/prometheus.go#L1864) by a prometheus metric. While the cdavisor is going over a loop of containers, there is a possibility that some of those containers might have been cleaned up from the system as well as the internal cache of the cadvisor. 

This problem had existed even earlier but it was fixed by this PR, https://github.com/google/cadvisor/pull/1985/ but only for the docker containers. That PR made sure that if for docker containers such cache miss happens instead of returning the error to the upper layers it will simply log it. For more info on the original issue, https://github.com/google/cadvisor/issues/1970

I simply had to add the same logic for other containers that already exists for the docker containers. So to maintain the consistency between the error messages I copied the error message returned by the [original implementation](https://github.com/google/cadvisor/pull/1985/files#diff-f679b61276b44dc3f4712950536707c0389ee4c536fc29837c0202c6e81c1bbfR633) for the docker containers.

Reply:
Got it!